lib/checker.js

```diff
@@ -1673,6 +1673,23 @@ function typesMatch(type1,type2) {
 	);
 }
 
+function isAssignmentAllowed(type1, type2){
```

Comment:
Could we rename these two parameters to `targetType` and `sourceType` so it's a little clearer?

Also, could we match the order `sourceType,targetType` that is used in several other functions? That includes changing the call to it on [line 1193](https://github.com/getify/Typl/pull/28/files#diff-605f23c7e3628b173a428e6a469e6ff6R1193).